File: ruby/spec/url.spec.rb
require 'enotype'

describe 'url' do
  context 'with \'http://www.valid.com\'' do
    it 'produces the expected result' do
      expect(Enotype::url('http://www.valid.com')).to eq('http://www.valid.com')
    end
  end
  
  context 'with \'https://valid.com\'' do
    it 'produces the expected result' do
      expect(Enotype::url('https://valid.com')).to eq('https://valid.com')
    end
  end
  
  context 'with \'https://www.valid.com\'' do
    it 'produces the expected result' do
      expect(Enotype::url('https://www.valid.com')).to eq('https://www.valid.com')
    end
  end
  
  context 'with \'invalid\'' do
    it 'raises an error' do
      expect { Enotype::url('invalid') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'www.invalid\'' do
    it 'raises an error' do
      expect { Enotype::url('www.invalid') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'www.invalid.com\'' do
    it 'raises an error' do
      expect { Enotype::url('www.invalid.com') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'htp://www.invalid.com\'' do
    it 'raises an error' do
      expect { Enotype::url('htp://www.invalid.com') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'http:/invalid.com\'' do
    it 'raises an error' do
      expect { Enotype::url('http:/invalid.com') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'https//invalid.com\'' do
    it 'raises an error' do
      expect { Enotype::url('https//invalid.com') }.to raise_error(RuntimeError)
    end
  end
  
  context 'with \'https://invalid\'' do
    it 'raises an error' do
      expect { Enotype::url('https://invalid') }.to raise_error(RuntimeError)
    end
  end
end

File: php/spec/url.spec.php
<?php declare(strict_types=1);

use Enotype\Loaders;

describe('url', function() {
  describe('with \'http://www.valid.com\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::url('http://www.valid.com'))->toEqual('http://www.valid.com');
    });
  });
  
  describe('with \'https://valid.com\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::url('https://valid.com'))->toEqual('https://valid.com');
    });
  });
  
  describe('with \'https://www.valid.com\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::url('https://www.valid.com'))->toEqual('https://www.valid.com');
    });
  });
  
  describe('with \'invalid\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::url('invalid'); })->toThrow();
    });
  });
  
  describe('with \'www.invalid\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::url('www.invalid'); })->toThrow();
    });
  });
  
  describe('with \'www.invalid.com\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::url('www.invalid.com'); })->toThrow();
    });
  });
  
  describe('with \'htp://www.invalid.com\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::url('htp://www.invalid.com'); })->toThrow();
    });
  });
  
  describe('with \'http:/invalid.com\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::url('http:/invalid.com'); })->toThrow();
    });
  });
  
  describe('with \'https//invalid.com\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::url('https//invalid.com'); })->toThrow();
    });
  });
  
  describe('with \'https://invalid\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::url('https://invalid'); })->toThrow();
    });
  });
});